### PR TITLE
Add CONTRIBUTING as to make our contributing docs more discoverable

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Creates the CONTRIBUTING document which links through to our existing contribution documents to help surface these in the various places it gets surfaced that [you can see here](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors).

This is more a discovery PR, there is a lot more we could do, this is to just get a start.

## Why?

As we have more and more contributors, it would be good to have a place for people to get an understanding of how to contribute more effectively, and hopefully break down barriers when contributing.

Also to help the platform team evolve these documents s the platform matures.

### CODE REVIEW

**Review by bryophyta — APPROVED** [review]
this is a great idea!

**Review by OllysCoding — APPROVED** [review]
Looks good to me, but would be great to expand this with more information soon!

**Comment by bryophyta** [comment]
we might want to add `CONTRIBUTING.md` to the `paths-ignore` rules for our GHA workflows, like we do for docs?

https://github.com/guardian/dotcom-rendering/blob/883f3422771854a57361d55ac11a7bfd45493732/.github/workflows/amp-validation.yml#L4-L6

Happy to raise this as a separate PR after this one's been merged though, if that's preferable.

File: dotcom-rendering/docs/contributing/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Contributing
+
+- [Detailed setup guide](./detailed-setup-guide.mddetail)
```
Comment (JamieB-gu) [inline dotcom-rendering/docs/contributing/README.md:3]:
```suggestion
- [Detailed setup guide](./detailed-setup-guide.md)
```

**Comment by jamesgorrie** [comment]
@JamieB-gu I've updated the readme to rather have starting points dependant on your information.

We could have a set of contributions documentation, and then technical jumping off points into the relevant projects?

**Comment by jamesgorrie** [comment]
> we might want to add `CONTRIBUTING.md` to the `paths-ignore` rules for our GHA workflows, like we do for docs?

@bryophyta I wonder if we make that a `**/*.md` instead (as part of another PR)?

File: apps-rendering/package-lock.json [hunk 2]
```diff
@@ -3376,7 +3376,7 @@
         "ms": {
```
Comment (JamieB-gu) [inline apps-rendering/package-lock.json:3376]: You may want to drop the changes to this file?

**Review by JamieB-gu — COMMENTED** [review]
> I've updated the readme to rather have starting points dependant on your information.

Looks great, thanks!

**Comment by bryophyta** [comment]
> @bryophyta I wonder if we make that a **/*.md instead (as part of another PR)?

yes, that sounds like a good idea!

**Comment by georgeblahblah** [comment]
I think that this is a great idea! I think there's a good opportunity to write some contributing docs for AR

File: apps-rendering/package-lock.json (hunk 2 above)
Comment (mxdvl) [reply apps-rendering/package-lock.json:3376]: Agreed!

**Pull request merged**